commands/addons/destroy.js

```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+const cli = require('heroku-cli-util')
+const co = require('co')
+const resolve = require('../../lib/resolve')
+const util = require('../../lib/util')
+const _ = require('lodash')
+
+function * run (context, heroku) {
+  let force = context.flags.force || process.env.HEROKU_FORCE === '1'
+  if (context.args.length === 0) {
+    throw new Error('Missing add-on name')
+  }
+
+  function showRelease (app, attachments) {
+    let names = attachments.map((att) => cli.color.configVar(att.name)).join(', ')
+    return cli.action(
+      `Removing config vars ${names} on ${cli.color.app(app)} and restarting`,
+      {success: false},
+      heroku.request({
+        path: `/apps/${app}/releases`,
+        partial: true,
+        headers: {Range: 'version ..; max=1, order=desc'}
+      })
+    ).then((release) => {
+      cli.console.error(`done, ${cli.color.release('v' + release[0].version)}`)
+    })
+  }
+
+  let destroy = co.wrap(function * (addon) {
+    let attachments = yield heroku.get(`/addons/${addon.name}/addon-attachments`)
+    let msg = `Destroying ${cli.color.addon(addon.name)} on ${cli.color.app(addon.app.name)}`
+    addon = yield cli.action(msg, {success: false}, heroku.request({
+      method: 'DELETE',
+      path: `/apps/${addon.app.id}/addons/${addon.id}`,
+      headers: {'Accept-Expansion': 'plan'},
+      body: {force}
+    }))
+    cli.console.error(addon.plan.price ? util.formatPrice(addon.plan.price) : '')
+
+    if (addon.config_vars.length > 0) {
+      for (let app of _.chain(attachments).groupBy('app.name').toPairs().value()) {
+        yield showRelease(app[0], app[1])
+      }
+    }
+  })
+
+  let addons = yield context.args.map((name) => resolve.addon(heroku, context.app, name))
```

Comment:
Yeah that's how it worked before

On Fri, Apr 29, 2016 at 2:38 PM Matte Noble <notifications@github.com>
wrote:

> In commands/addons/destroy.js
> <https://github.com/heroku/heroku-cli-addons/pull/30#discussion_r61648513>
> :
>
> > +    addon = yield cli.action(msg, {success: false}, heroku.request({
> > +      method: 'DELETE',
> > +      path: `/apps/${addon.app.id}/addons/${addon.id}`,
> > +      headers: {'Accept-Expansion': 'plan'},
> > +      body: {force}
> > +    }))
> > +    cli.console.error(addon.plan.price ? util.formatPrice(addon.plan.price) : '')
> > +
> > +    if (addon.config_vars.length > 0) {
> > +      for (let app of _.chain(attachments).groupBy('app.name').toPairs().value()) {
> > +        yield showRelease(app[0], app[1])
> > +      }
> > +    }
> > +  })
> > +
> > +  let addons = yield context.args.map((name) => resolve.addon(heroku, context.app, name))
>
> Does this mean you can pass multiple identifiers to delete add-on en batch?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/heroku/heroku-cli-addons/pull/30/files/aa8ca55dec56d29ce19860ba4806a0031559c378#r61648513>
>